File: AcademicAPI/Models/Kinder/Alumno.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

#nullable disable

namespace AcademicAPI.Models.Kinder
{
    public partial class Alumno
    {
        public Alumno()
        {
            Asistencia = new HashSet<Asistencium>();
            Pagos = new HashSet<Pago>();
        }

        public int IdAlumno { get; set; }
        public string Nombres { get; set; }
        public string ApellidoPaterno { get; set; }
        public string ApellidoMaterno { get; set; }
        public string Curp { get; set; }
        public int Aula { get; set; }
        public bool Activo { get; set; }

        public virtual Aula AulaNavigation { get; set; }
        
        [JsonIgnore]
        public virtual ICollection<Asistencium> Asistencia { get; set; }

        [JsonIgnore]
        public virtual ICollection<Pago> Pagos { get; set; }
    }
}

File: React.UI/ClientApp/src/models/Alumno.ts
import {aulaSchema} from "./Aula";
import { z } from "zod";
import {curp, numeric, text, id} from "./Schema";

const alumnoSchema = z.object({
  idAlumno: id,
  nombres: text,
  apellidoPaterno: text,
  apellidoMaterno: text,
  curp: curp,
  aula: numeric,
  activo: z.boolean().optional().default(true),
  aulaNavigation: aulaSchema.optional(),
  presente: z.boolean().optional(),
  isDeudor: z.boolean().optional(),
});

type IAlumno = z.TypeOf<typeof alumnoSchema>;

export type { IAlumno };
export { alumnoSchema };
